pilot/pkg/model/push_context.go

```diff
@@ -800,36 +798,44 @@ func (ps *PushContext) GatewayServices(proxy *Proxy) []*Service {
 	return gwSvcs
 }
 
+func (ps *PushContext) ServiceAttachedToGateway(hostname string, proxy *Proxy) bool {
+	for _, gw := range proxy.MergedGateway.GatewayNameForServer {
+		if hosts := ps.virtualServiceIndex.destinationsByGateway[gw]; hosts != nil {
+			if _, f := hosts[hostname]; f {
+				return true
+			}
+		}
+	}
+	return false
+}
+
 // add services from MeshConfig.ExtensionProviders
 // TODO: include cluster from EnvoyFilter such as global ratelimit [demo](https://istio.io/latest/docs/tasks/policy-enforcement/rate-limit/#global-rate-limit)
-func getHostsFromMeshConfig(ps *PushContext) sets.String {
-	hostsFromMeshConfig := sets.New[string]()
-
+func addHostsFromMeshConfig(ps *PushContext, hosts map[string]sets.Set[int]) {
 	for _, prov := range ps.Mesh.ExtensionProviders {
 		switch p := prov.Provider.(type) {
 		case *meshconfig.MeshConfig_ExtensionProvider_EnvoyExtAuthzHttp:
-			hostsFromMeshConfig.Insert(p.EnvoyExtAuthzHttp.Service)
+			hosts[p.EnvoyExtAuthzHttp.Service] = nil
```

Comment:
I don't get it, why are we setting to nil? doesn't that erase all the ports we have in the inner set?

Reply:
Because we don't care about the ports, we only need to include these hosts in `destinationsByGateway`